src/averageAges.js

```diff
@@ -37,7 +42,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter((p) => p.sex === 'f'
+      && people.find((p1) => p1.mother === p.name))
+    : people.filter((p) => p.sex === 'f');
```

Comment:
1. It is better to avoid names with one letter, in this case you can use the destructuring 
https://mate-academy.github.io/style-guides/javascript.html#51-use-object-destructuring-when-accessing-and-using-multiple-properties-of-an-object 
2. You can reduce the number of operations by creating an object womenWithChildrenMap, where the keys are the names of women who have a child, define all mothers in one pass, and then use it as a condition: 
```
const women = withChildren
    ? people.filter(({ name, sex }) => sex === 'f'
      && womenWithChildrenMap[name])
    : people.filter(({ sex }) => sex === 'f');
```